### PR TITLE
LPS-77925

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
For more information, see: https://issues.liferay.com/browse/LPS-77925

/cc: @marianoalvarosaiz

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test

**Comment by achaparro** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jorgediaz-lr** [comment]
👀

**Comment by jorgediaz-lr** [comment]
👀

**Comment by jorgediaz-lr** [comment]
👀

File: modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ */
+public class UpgradeJournalArticleDates extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_updateCreateDate();
+		_updateModifiedDate();
+	}
+
+	private void _updateCreateDate() throws SQLException {
```
Comment (achaparro) [inline modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: Hi guys,

What we want in this method is to set the createDate for the first version in the versions related to that article. I guess we can do this in just one update clause. What do think? @marianoalvarosaiz @jorgediaz-lr

**Comment by jorgediaz-lr** [comment]
👀

File: modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java (hunk 1 above)
Comment (achaparro) [reply modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: If you don't agree with this change at least I think that we can remove the usage of JournalArticleResource in the first query.

Comment (jorgediaz-lr) [reply modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: I agree in removing JournalArticleResource table from SQL, but I think it doesn't worth joining select and update sqls.

Comment (achaparro) [reply modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: I have a proposal for you @marianoalvarosaiz and @jorgediaz-lr, we see it tomorrow

Comment (marianoalvarosaiz) [reply modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: We shouldn't do an update with subselect due to MySQL optimization shortcomings: https://dev.mysql.com/doc/refman/5.7/en/subquery-optimization.html

Comment (achaparro) [reply modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java]: :thumbsup: :heart_eyes:

**Comment by achaparro** [comment]
Pull request submitted to https://github.com/achaparro/liferay-portal/pull/129. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jorgediaz-lr** [comment]
👀 👀

**Comment by achaparro** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/56079. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**